cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/dataset/MetadataDataset.java

```diff
@@ -79,7 +79,92 @@
 import javax.annotation.Nullable;
 
 /**
- * Dataset that manages Metadata using an {@link IndexedTable}.
+ * Dataset that manages Metadata using an {@link IndexedTable}. Supports writing tags and properties. Tags are single
+ * elements, whereas properties are key-values.
+ *
+ * Tags and properties both get translated into a {@link MetadataEntry}. A {@link MetadataEntry} corresponds to all
+ * metadata that is set on some {@link MetadataEntity}, like an application or a namespace. Each entry contains the
+ * entity the metadata is for, a key, value, and schema. A metadata property is represented as a MetadataEntry
+ * where the key is the property key and the value is the property value. Metadata tags are represented as a
+ * MetadataEntry where the key is 'tags' and the value is a comma separated list of the tags.
+ *
+ * The value for a {@link MetadataEntry} is stored under the 'v' column as a string. The data here is not indexed.
+ * The row key used to store metadata entries is a composite MDSKey. It always begins with:
+ *
+ * v:[entity-type]
```

Comment:
lets say metadata entity type since it can be custom entity too.

Reply:
done